test/cmake/diagnostics/list_missing_unit_tests.cmake

```diff
@@ -0,0 +1,45 @@
+# -----------------------------------------------------------------------------------------------------
+# Copyright (c) 2006-2020, Knut Reinert & Freie Universität Berlin
+# Copyright (c) 2016-2020, Knut Reinert & MPI für molekulare Genetik
+# This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+# shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+# -----------------------------------------------------------------------------------------------------
+
+set(seqan3_test_include_targets "" CACHE STRING "" FORCE)
+
+function (collect_include_target include_target)
+    set(seqan3_test_include_targets "${seqan3_test_include_targets};${include_target}" CACHE STRING "" FORCE)
+endfunction ()
+
+function (list_missing_unit_tests)
+    if (CMAKE_VERSION VERSION_LESS 3.8) # MANUALLY_ADDED_DEPENDENCIES since cmake >= 3.8
+        return ()
+    endif ()
```

Comment:
Could you instead emit a warning that you need cmake >= 3.8 for this feature?

Reply:
I just saw below
```cmake
if (SEQAN3_USE_INCLUDE_DEPENDENCIES)
    cmake_minimum_required (VERSION 3.8)
endif ()
```
So I think it would be helpful to have a warning if you somehow end up in this function even though `cmake_minimum_required` should have stopped you.